packages/testcontainers/src/wait-strategies/http-wait-strategy.ts

```diff
@@ -107,9 +107,36 @@ export class HttpWaitStrategy extends AbstractWaitStrategy {
       this.startupTimeout
     );
 
+    await Promise.race([this.waitContainerNotExited(container), healthCheckPromise]);
+
     log.debug(`HTTP wait strategy complete`, { containerId: container.id });
   }
 
+  private async waitContainerNotExited(container: Dockerode.Container) {
+    const exitStatus = 'exited';
+    const timeout = this.startupTimeout + 1000; // added 1 sec to avoid race condition
+    const status = await new IntervalRetry<string, null>(500).retryUntil(
+      async () => (await container.inspect()).State.Status,
+      (status) => status === exitStatus,
+      () => null,
+      timeout
+    );
+
+    if (status === null) {
+      return;
+    }
+
+    if (status === exitStatus) {
+      const tail = 50;
+      const lastLogs = (await container.logs({ tail, stdout: true, stderr: true })).toString();
+      const message = `container exited, last ${tail} logs: ${lastLogs}`;
+
+      log.error(message, { containerId: container.id });
```

Comment:
I think it can be warning/debug, if we not throw error after it (shutdownOnExit=false)